src/clib/pio_darray.c

```diff
@@ -825,30 +856,30 @@ static void *PIOc_convert_buffer_adios(file_desc_t *file, io_desc_t *iodesc,
     memcpy(temp_buf, array, sizeof(var_type)); \
 }
 
-void *PIOc_copy_one_element_adios(void *array, adios_var_desc_t *av)
+void *PIOc_copy_one_element_adios(void *array, io_desc_t *iodesc)
```

Comment:
Asserts needed at the beginning of the function to ensure that the input args are not NULL